### PR TITLE
feat(plasma-tokens-native): Add generate native tokens and themes in new package

### PR DESCRIPTION
**Pull request by neretin-trike**
Методы генерации схем для нативных платформ (т.е. содержимое папки `properties`)  были перенесены в библиотеку `plasma-tokens-utils`.

Методы генерации тем (токенов) для нативных платформ (т.е. содержимое папки `build`) были перенесены в новую библиотеку `plasma-tokens-native`

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.210.0-canary.494.4991833636.0
  npm install @salutejs/plasma-core@1.115.0-canary.494.4991833636.0
  npm install @salutejs/plasma-hope@1.210.0-canary.494.4991833636.0
  npm install @salutejs/plasma-icons@1.144.0-canary.494.4991833636.0
  npm install @salutejs/plasma-temple@1.159.0-canary.494.4991833636.0
  npm install @salutejs/plasma-tokens-b2b@1.18.0-canary.494.4991833636.0
  npm install @salutejs/plasma-tokens-b2c@0.27.0-canary.494.4991833636.0
  npm install @salutejs/plasma-tokens-web@1.33.0-canary.494.4991833636.0
  npm install @salutejs/plasma-tokens@1.50.0-canary.494.4991833636.0
  npm install @salutejs/plasma-ui@1.191.0-canary.494.4991833636.0
  npm install @salutejs/plasma-web@1.210.0-canary.494.4991833636.0
  npm install @salutejs/plasma-cy-utils@0.57.0-canary.494.4991833636.0
  npm install @salutejs/plasma-sb-utils@0.113.0-canary.494.4991833636.0
  npm install @salutejs/plasma-tokens-utils@0.24.0-canary.494.4991833636.0
  # or 
  yarn add @salutejs/plasma-b2c@1.210.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-core@1.115.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-hope@1.210.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-icons@1.144.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-temple@1.159.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-tokens-b2b@1.18.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-tokens-b2c@0.27.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-tokens-web@1.33.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-tokens@1.50.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-ui@1.191.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-web@1.210.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-cy-utils@0.57.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-sb-utils@0.113.0-canary.494.4991833636.0
  yarn add @salutejs/plasma-tokens-utils@0.24.0-canary.494.4991833636.0
  ```
</details>

:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.494.4991833636.swift)  
[color_sberHealth_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.494.4991833636.kt)  
[color_sberHealth_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.494.4991833636.ts)  
[color_sbermarket_business_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.494.4991833636.swift)  
[color_sbermarket_business_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.494.4991833636.kt)  
[color_sbermarket_business_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.494.4991833636.ts)  
[color_sbermarket_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.494.4991833636.swift)  
[color_sbermarket_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.494.4991833636.kt)  
[color_sbermarket_metro_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.494.4991833636.swift)  
[color_sbermarket_metro_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.494.4991833636.kt)  
[color_sbermarket_metro_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.494.4991833636.ts)  
[color_sbermarket_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.494.4991833636.ts)  
[color_sbermarket_selgros_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.494.4991833636.swift)  
[color_sbermarket_selgros_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.494.4991833636.kt)  
[color_sbermarket_selgros_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.494.4991833636.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.494.4991833636.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.494.4991833636.kt)  
[color_sbermarket_wlbusiness_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.494.4991833636.ts)  
[color_sberprime_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.494.4991833636.swift)  
[color_sberprime_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.494.4991833636.kt)  
[color_sberprime_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.494.4991833636.ts)  
[shadow_sbermarket_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.494.4991833636.ts)  
[typo_mage_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.494.4991833636.swift)  
[typo_mage_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.494.4991833636.kt)  
[typo_mage_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.494.4991833636.ts)  
[typo_plasma_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.494.4991833636.swift)  
[typo_plasma_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.494.4991833636.kt)  
[typo_plasma_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.494.4991833636.ts)  
[typo_ruler_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.494.4991833636.swift)  
[typo_ruler_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.494.4991833636.kt)  
[typo_ruler_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.494.4991833636.ts)  
[typo_sage_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.494.4991833636.swift)  
[typo_sage_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.494.4991833636.kt)  
[typo_sage_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.494.4991833636.ts)  
[typo_sbermarket_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.494.4991833636.swift)  
[typo_sbermarket_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.494.4991833636.kt)  
[typo_sbermarket_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.494.4991833636.ts)  
[typo_soulmate_ios-swift--canary.494.4991833636.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.494.4991833636.swift)  
[typo_soulmate_kotlin--canary.494.4991833636.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.494.4991833636.kt)  
[typo_soulmate_react-native--canary.494.4991833636.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.494.4991833636.ts)

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-494/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-494/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-494/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-494/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-494/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-494/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-494/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-494/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-494/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-494/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-494/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-494/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-494/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-494/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-494/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-494/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-494/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

File: utils/plasma-tokens-utils/src/generators/generateColorThemesTokenDataGroups.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    return {
+        [name]: {
+            [`${name}__dark`]: dataObjectToTokenDataGroup((dark as unknown) as DataObject, ''),
+            [`${name}__light`]: dataObjectToTokenDataGroup((light as unknown) as DataObject, ''),
+        },
+    };
+};
+
+export const generateColorThemesTokenDataGroups = (themesFolder: string): Record<string, ThemeTokenDataGroups> => {
+    if (!fs.existsSync(themesFolder)) {
```
Comment (Yakutoc) [inline utils/plasma-tokens-utils/src/generators/generateColorThemesTokenDataGroups.ts:53]: @neretin-trike Что думаешь на это счет - метод `existsSync` deprecated? 

https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback
https://nodejs.org/api/fs.html#fs_fs_exists_path_callback

File: utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    tertiary: 'textTertiary',
+    paragraph: 'textParagraph',
+    background: 'backgroundPrimary',
+    accent: 'textAccent',
+    success: 'textPositive',
+    warning: 'textWarning',
+    critical: 'textNegative',
+    overlay: 'overlaySoft',
+
+    // gradient = 'Градиент для заливки основного фона',
```
Comment (Yakutoc) [inline utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts]: @neretin-trike Почему закомментировано? Что-то служебное?

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

File: utils/plasma-tokens-native/src/generate.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    sbermarketTypoProperties,
+} from '@salutejs/plasma-typo';
+
+// Директория создаётся после выполнения команды prepare
+import { sbermarketShadows } from '../data/shadows';
+
+import { createProperties } from './creators';
+
+const outDir = 'src';
+fs.existsSync(outDir) || fs.mkdirSync(outDir);
```
Comment (Yakutoc) [inline utils/plasma-tokens-native/src/generate.ts:19]: @neretin-trike У тебя же уже есть lib которая проверяет/создает директорию?

File: packages/plasma-tokens/lib/creators/createRootIndex.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import fs from 'fs';
+import path from 'path';
+import { ROBO_COMMENT, HTML_FONT_SIZE } from '@salutejs/plasma-tokens-utils';
+
+import { typo } from '../../data';
+
+export const createRootIndex = (srcDir: string) => {
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/createRootIndex.ts:7]: не, это не про нейтив, тут я просто переместил на уровень выше

File: utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts (hunk 2 above)
Comment (neretin-trike) [reply utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts]: тут предполагается, что мы когда-нибудь (а может и нет) сделаем маппинг на все старые токены. Ну и ещё это как список тех, которые не замплены, чтобы иметь их ввиду

File: utils/plasma-tokens-native/src/generate.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    mageTypoProperties,
+    sageTypoProperties,
+    soulmateTypoProperties,
+    rulerTypoProperties,
+    plasmaTypoProperties,
+    sbermarketTypoProperties,
+} from '@salutejs/plasma-typo';
+
+// Директория создаётся после выполнения команды prepare
+import { sbermarketShadows } from '../data/shadows';
```
[the remark this answers is not in the document]
Comment (Yakutoc) [reply utils/plasma-tokens-native/src/generate.ts:14]: @Yeti-or А накинь пример как ты видишь?

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-494/

File: packages/plasma-tokens/lib/creators/createBrands.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import path from 'path';
+import {
+    generateTokens,
+    writeGeneratedToFS,
+    lowerFirstLetter,
+    ThemeTokenDataGroups,
+} from '@salutejs/plasma-tokens-utils';
+
+export const createBrands = (srcDir: string, themesColorTokenGroupsFallback: ThemeTokenDataGroups) => {
+    const fixedThemeType = '__dark';
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: в этом методе генерится два файла: `index.ts` и `value.ts` В первом просто набор токенов (js переменных), в котором в качестве значение css переменные типа:

```
export const buttonCritical = 'var(--plasma-colors-button-critical, #FF293E)';
```

Во втором (value.ts) лежат уже обычные строковые значения, но только для dark

```
export const buttonCritical = '#FF293E';
```

Тут на самом деле у меня сомнения счёт того, надо ли вообще в таком случае файл value.ts генерить

File: utils/plasma-tokens-utils/src/generators/generateColorThemesTokenDataGroups.ts (hunk 1 above)
Comment (neretin-trike) [reply utils/plasma-tokens-utils/src/generators/generateColorThemesTokenDataGroups.ts:53]: хм, у меня не IDE не подсвечивает, что `existsSync` deprecated. В принципе могу и Васин вариант использовать

File: packages/plasma-tokens/package.json [hunk 7]
```diff
@@ -49,21 +42,14 @@
         "@salutejs/plasma-colors": "0.6.0",
         "@salutejs/plasma-tokens-utils": "0.22.0",
         "@salutejs/plasma-typo": "0.33.0",
-        "@types/inquirer": "8.2.1",
+        "@types/fs-extra": "11.0.1",
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/package.json]: по дефолту берётся из ноды, но она не тепизирована, поэтому приходится ставить ещё типы

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-494/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-494/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-494/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-494/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-494/temple-storybook/

File: utils/plasma-tokens-native/package.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
+{
+    "name": "@salutejs/plasma-tokens-native",
+    "version": "1.0.0",
+    "description": "Plasma tokens for swift, kotlin, react-native",
+    "module": "es/lib/index.js",
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply utils/plasma-tokens-native/package.json:5]: нет, он просто упаковывает файлики в папку, как это было с plasma-tokens-android / ios-swift:  https://www.npmjs.com/package/@salutejs/plasma-tokens-native?activeTab=code

File: packages/plasma-tokens/lib/creators/createBrands.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: я чет запутался опять, это же не про натив речь?

File: packages/plasma-tokens/lib/creators/createRootIndex.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/createRootIndex.ts:7]: а ок

File: utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    blackPrimary: 'onLightTextPrimary',
+    blackSecondary: 'onLightTextSecondary',
+    blackTertiary: 'onLightTextTertiary',
+
+    transparent: 'clear',
+
+    buttonClear: 'clear',
+    buttonBlack: 'onLightSurfaceSolidDefault',
+    buttonBlackSecondary: 'onLightSurfaceTransparentSecondary',
+    // buttonBlackTransparent = 'Чёрная для использования поверх картинок',
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply utils/plasma-tokens-utils/src/generators/getMapDeprecatedColorTokens.ts]: вот [тут](https://github.com/salute-developers/plasma/pull/494#discussion_r1180277405) Саше отвечал

File: packages/plasma-tokens/lib/creators/createBrands.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: а я ниже прочитал, хз кажется  value становится бесполезным в таком случае да

[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: но это вроде как ломает обратную совместимость? если дропнуть

File: utils/plasma-tokens-native/src/generate.ts (hunk 3 above)
Comment (neretin-trike) [reply utils/plasma-tokens-native/src/generate.ts:19]: кстати справедливо, не обратил внимание, что это можно удалить. Спасибо 👌

File: packages/plasma-tokens/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/package.json]: так это внешний модуль, он не может из ноды браться ) просто он у нас в корне лежит
https://github.com/salute-developers/plasma/blob/master/package.json#L95
и если там что-то поменятся то у тебя внезапно развалится сборка

File: utils/plasma-tokens-native/src/generate.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply utils/plasma-tokens-native/src/generate.ts:14]:
```
import *  as shadows from '../data/shadows';

Object.entries(shadows).forEach(([key, value])
```

File: utils/plasma-tokens-native/src/generate.ts (hunk 3 above)
Comment (Yeti-or) [reply utils/plasma-tokens-native/src/generate.ts:19]: fs.existsSync(outDir) || fs.mkdirSync(outDir); => fs.ensureDir(outDir)

File: packages/plasma-tokens/lib/creators/createBrands.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: честно говоря сомневаюсь, что кто-то в своих проектах вот такой путь для получения строкового значения сейчас использует:  `import { buttonCritical } from '@salutejs/plasma-tokens/brands/sbermarket_metro/values'`. 
Выглядит прям очень стремно, предлагаю тогда её не собирать (убери в прке с тенями, которую сейчас делаю), но если вдруг у пользователей появится необходимость получать именно значения, то подумаем как это можно сделать для двух режимов (dark / light)

[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/lib/creators/createBrands.ts:11]: окей давай так да

File: utils/plasma-tokens-native/package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply utils/plasma-tokens-native/package.json:5]: ну тогда ему не нужны ни main ни module ни sideEffects

File: utils/plasma-tokens-native/package.json [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    "main": "lib/index.js",
+    "types": "lib/index.d.ts",
+    "files": ["build"],
```
Comment (Yeti-or) [inline utils/plasma-tokens-native/package.json:8]: а сам build кстати нужен?

вообще весь пакет нужно ли запихивать в npm?
если нам от него надо только assets

File: utils/plasma-tokens-native/src/generate.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply utils/plasma-tokens-native/src/generate.ts:14]: кажется, что тут вообще можно будет брать shadow прямо из json, как я сейчас делаю для теней.  У меня в будущей прке такая сейчас структура:

sbermarket.json:
```
{
   light: {
      ...tokens
   },
   dark: {
      ...tokens
   }
   shadow: {
      ...tokens
   }
}
```

Т.е. я предлагаю сделать это следующими итерациями

File: utils/plasma-tokens-native/src/generate.ts (hunk 3 above)
Comment (neretin-trike) [reply utils/plasma-tokens-native/src/generate.ts:19]: да, так сейчас и есть но в пакете plsama-tokens :) Тут вообще эти создания не нужны

File: utils/plasma-tokens-native/package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply utils/plasma-tokens-native/package.json:5]: да, это мой косяк, что я их добавил, когда экспериментировал и забыл убрать, спасиб 👌

File: utils/plasma-tokens-native/package.json (hunk 10 above)
Comment (neretin-trike) [reply utils/plasma-tokens-native/package.json:8]: кажется, что только build и нужен:
https://www.npmjs.com/package/@salutejs/plasma-tokens-native?activeTab=code

File: packages/plasma-tokens/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/package.json]: блин :/ просто `npm list ts-extra` не показывает откуда модуль пришёл (как и для нодовых зависимостей), вот и я подумал. А нам точно тогда надо доп зависимость ставить для такой не сильно значимой штуки?

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: utils/plasma-tokens-native/package.json (hunk 10 above)
Comment (Yeti-or) [reply utils/plasma-tokens-native/package.json:8]: А зачем нам билд?

Comment (neretin-trike) [reply utils/plasma-tokens-native/package.json:8]: ну там же в итоге лежат артифакты, которые можно забрать. Или ты что-то другое имеешь ввиду? Не совсем понял тогда

**Comment by neretin-trike** [comment]
> чтобы не забыть вопрос - а точно ли нам нужен пакет plasma-tokens-native в `npm` можем ли мы его не публиковать

а как тогда с него забирать файлы тем? Или ты предлагаешь оставить их только в релизе и всё?

File: packages/plasma-tokens/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (Yeti-or) [reply packages/plasma-tokens/package.json]: точно

File: utils/plasma-tokens-native/package.json (hunk 10 above)
Comment (Yeti-or) [reply utils/plasma-tokens-native/package.json:8]: эти артифакты у нас не в assets лежат? а именно в npm ?

Comment (neretin-trike) [reply utils/plasma-tokens-native/package.json:8]: ну сейчас и в assets и в npm. Как и было с plasma-tokens-android / ios

File: packages/plasma-tokens/package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-tokens/package.json]: решили в итоге удалить эту зависимость и оставить прежнюю реализацию: ` fs.existsSync(brandsDir) || fs.mkdirSync(brandsDir);`

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-494/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-494/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: package.json [hunk 11]
```diff
@@ -125,7 +125,7 @@
         "upload-assets",
         {
           "assets": [
-            "./packages/plasma-tokens/build/**/*"
```
Comment (Yeti-or) [inline package.json:128]: 👍

File: utils/plasma-tokens-native/README.md [hunk 12]
```diff
@@ -0,0 +1,24 @@
+# Plasma-tokens-native
+
+Пакет предоставляет набор `дизайн-токенов` реализующих дизайн «Plasma».
+
+Это служебный пакет с реализацией под под платформы `kotlin`, `react-native`, `swift` сгенерированный из основного [plasma-tokens](https://github.com/salute-developers/plasma/tree/master/packages/plasma-tokens).
```
Comment (Yeti-or) [inline utils/plasma-tokens-native/README.md:5]: может докинуть сюда инфу о том что артифакты можно забрать из релизов?

**Review by Yeti-or — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-494/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-494/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-494/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-494/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-494/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-494/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**